src/util/nodes.tsx

```diff
@@ -74,6 +77,43 @@ export const parseBranch = (inputString: string): TreeNode | null => {
   }
 }
 
+const replaceSymbols = (inputString: string): string => {
+  // const singleArrow = inputString.replace('->', '⊃') // replace -> with proper hook symbol
+  // const doubleArrow = singleArrow.replace('=>', '⊃') // replace => with proper hook symbol
+  // const cond = doubleArrow.replace('>', '⊃') // replace > with proper hook symbol
+  // const negation = cond.replace('~', '¬') // replace ~ with proper negation symbol
+  // const dash = negation.replace('-', '¬') // replace - with proper negation symbol
+  // const conjunction = dash.replace('^', '&') // replace ^ with proper conjunction symbol
+  // const disjunction = conjunction.replace('v', '∨') // replace v with proper disjunction symbol
+  // const final = disjunction.replace('=', '≡') // replace = with proper biconditional symbol
+
+  const initialValue = inputString
+
+  const symbols = {
+    '->': '⊃',
+    '=>': '⊃',
+    '>': '⊃',
+    '~': '¬',
+    '-': '¬',
+    '^': '&',
+    '=': '≡',
+    v: '∨',
+    '/': '∨',
+    '/\\': '&',
+  }
+
+  const newInput = Object.entries(symbols).reduce(reducer, initialValue)
+  return newInput
+}
+
+const reducer = (
+  inputString: string,
+  [inside, out]: [string, string]
```

Comment:
the name "inside" doesn't really make sense here. 'in' (which I assume you couldn't use because it's a typescript keyword) was implicitly short for 'input'.